ui/src/services/csrfTokenApi.ts

```diff
@@ -131,4 +137,21 @@ export function setupCsrfInterceptor(csrfTokenApi: CreateActions<CsrfTokenApi>,
 
         return config
     })
+
+    axiosInstance.interceptors.response.use(undefined, async (error: AxiosError) => {
+        if (isAxiosError(error) && csrfTokenApi.isInvalidCsrfToken(error)) {
+            const config = error.config
+
+            if (config && config.headers['Prefect-Csrf-Retry-Count']) {
+                const retryCount = parseInt(config.headers['Prefect-Csrf-Retry-Count'], 10)
+                if (retryCount < MAX_RETRIES) {
+                    await csrfTokenApi.addCsrfHeaders(config)
+                    config.headers['Prefect-Csrf-Retry-Count'] = (retryCount + 1).toString()
```

Comment:
Just double checking - this header is echoed back by the API, right? Or is the intent that if this fails a second time it'll have the same axios instance and so the header will be present from config (assuming this has run previously)

Reply:
It's no echoed back by the API. The `config` here is the request we initially fired that resulted in the error. So we're essentially just using the headers as data storage.

Reply:
Ok got it, so it's the second - that makes more sense, thank you